examples/rust/examples/maker/maker.rs

```diff
@@ -0,0 +1,732 @@
+use crate::settings::Settings;
+use ethers::abi::{AbiEncode, RawLog};
+use ethers::prelude::{
+    rand::{thread_rng, Rng},
+    Address, EthLogDecode, Http, JsonRpcClient, LocalWallet, Middleware, Provider, Signer,
+    SignerMiddleware, Ws, U256,
+};
+use ethers::utils::keccak256;
+use http::Uri;
+use siwe::{TimeStamp, Version};
+use std::ops::Mul;
+use std::{env, process::exit, sync::Arc};
+use time::OffsetDateTime;
+use tokio::sync::mpsc;
+use tonic::transport::{Channel, ClientTlsConfig};
+use valorem_trade_interfaces::bindings;
+use valorem_trade_interfaces::grpc_codegen;
+use valorem_trade_interfaces::grpc_codegen::{auth_client::AuthClient, Empty, VerifyText};
+use valorem_trade_interfaces::grpc_codegen::{
+    rfq_client::RfqClient, Action, ConsiderationItem, EthSignature, ItemType, OfferItem, Order,
+    OrderType, QuoteRequest, QuoteResponse, SignedOrder, H256,
+};
+use valorem_trade_interfaces::utils::session_interceptor::SessionInterceptor;
+
+mod settings;
+
+const SESSION_COOKIE_KEY: &str = "set-cookie";
+
+/// An example Market Maker (MM) client interface to Quay.
+///
+/// The Market Maker will receive Request For Quote (RFQ) from the Quay server formatted as
+/// `QuoteRequest` and the MM needs to respond with `QuoteResponse`.
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
+    if args.len() != 1 {
+        eprintln!("Unexpected command line arguments. Received {:?}", args);
+        eprintln!("Usage: maker <settings_file>");
+        exit(1);
+    }
+
+    let settings = Settings::load(&args[0]);
+
+    if settings.node_endpoint.starts_with("http") {
+        let provider = Provider::<Http>::try_from(settings.node_endpoint.clone())?;
+        run(Arc::new(provider), settings).await;
+    } else if settings.node_endpoint.starts_with("ws") {
+        // Websockets (ws & wss)
+        let provider = Provider::<Ws>::new(Ws::connect(settings.node_endpoint.clone()).await?);
+        run(Arc::new(provider), settings).await;
+    } else {
+        // IPC
+        let provider = Provider::connect_ipc(settings.node_endpoint.clone()).await?;
+        run(Arc::new(provider), settings).await;
+    }
+
+    unreachable!("Run function has returned from a while true loop");
+}
+
+// Main execution function. The Maker will execute the following use case.
+// 1. Connect and authorise itself with Quay
+// 2. Wait for an RFQ from the Taker
+// 3a. If the internal "randomness" counter is divisible by 3 and the order was a buy, the Maker will:
+//     Not offer a quote and move to step 2.
+// 3b. If the internal "randomness" counter is not divisible by 3 or the order was a sell, the Maker will:
+//     If the order was buy - Write to the Settlement contract for the amount of options
+//     (sourced from the RFQ) against the Option Type (sourced by the RFQ)
+//     Return a quote to the Taker.
+//     Move to step 2.
```

Comment:
```suggestion
// Main execution function.
```
Given the logic changed around the hardcode line of no quotes, I would just remove this description altogether